libgnucash/engine/Account.h

```diff
@@ -1454,6 +1454,11 @@ void gnc_account_delete_map_entry (Account *acc, char *head, char *category,
  */
 void gnc_account_delete_all_bayes_maps (Account *acc);
 
+/** Remove color slots that have a "Not Set" value
+ *  This should only be run once on a book
+ */
+void gnc_account_fix_color_slots (QofBook *book);
```

Comment:
This would fit best as part of xaccAccountScrubKvp except that that isn't presently called in the SQL backend. Generally the scrubs need some cleanup; calls to them are splattered around instead of consolidated and some of them are probably obsolete. To facilitate including this in an eventual cleanup let's call it xaccAccountScrubColorNotSet.

Reply:
Yours needs to run every session too in case a user edits an account with an older version of GnuCash.

I'll merge it and make any more required changes before the release.